src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -132,30 +132,30 @@ func DeduceBrokers(
 			return []string{"127.0.0.1:9092"}
 		}
 
-		if len(conf.Redpanda.KafkaApi) == 0 {
+		if len(conf.Redpanda.AdvertisedKafkaApi) == 0 {
```

Comment:
don't you need a fallback ? 

1. do the rpk thing you proposed. if missing: do 2
2. check if advertised api is defined, if missing do: 3
3. use kafka_api if missing, do 4
4. return error

Reply:
i don't see how this would work for most deployments.

Reply:
I sent this before I got the green light on the new rpk config structure. Reverting to draft.